sdks/python/apache_beam/transforms/window.py

```diff
@@ -164,14 +168,19 @@ def merge(self, merge_context):
     raise NotImplementedError
 
   def is_merging(self):
+    # type: () -> bool
+
     """Returns whether this WindowFn merges windows."""
     return True
 
   @abc.abstractmethod
   def get_window_coder(self):
+    # type: () -> coders.Coder
```

Comment:
This should be a `Coder[BoundedWindow]`.  We could even make WindowFn parameterized by the window type, and restrict this further.

Reply:
`coder.Coder` is not yet `Generic`, so we can't do that just yet, though we can do this in a future PR.  So this should not be one of the `*WindowCoder` types?